packages/big-design/src/components/StatefulTable/reducer.ts

```diff
@@ -66,7 +74,13 @@ export type Action<T> =
   | { type: 'SELECTED_ITEMS'; selectedItems: T[] }
   | { type: 'SORT'; column: StatefulTableColumn<T>; direction: TableSortDirection }
   | { type: 'SET_PILL_TABS_PROPS'; pillTabsProps: PillTabsProps }
-  | { type: 'TOGGLE_PILL'; pillId: string; filter: StatefulTablePillTabFilter<T>['filter'] };
+  | { type: 'SEARCH_VALUE_CHANGE'; value: string }
+  | {
+      type: 'ON_FILTER_ITEMS';
+      filterSearch?: StatefulTableSearch<T>['filter'];
+      filterPills?: StatefulTablePillTabFilter<T>['filter'];
+      pillId?: string;
+    };
```

Comment:
Couple of things here,
1. We probably shouldn't couple filters with search, as in, they are not synonymous. It also seems like if you have a partial search value without pressing search, then click on a pill, you'll get inaccurate results (my intention is to filter by low stock after forgetting to clear the search field):
![Screen Shot 2021-04-19 at 15 49 05](https://user-images.githubusercontent.com/10539418/115301458-d936ab80-a126-11eb-98f8-c59d5b5413d0.png)
2. Search in a stateful component means many different things, not just searching on the name value. For example, what if I wanted to search on SKU? I feel like the API needs to be flexible to search among many fields.

Reply:
1. Hm, I partially agree with you. But I couldn't image how can we split filters and search. Users can have opportunity to filter Table with PillTabs and search at the same time (or not ? 🧐). 
So we should filter data with PillTabs and with search. It's happening step by step and we should save previous step and values. 

Maybe add `isClickedSearchSubmit` for this case? 

2. You can choose it in  the `filter` function in the search and filter by `sku` 
```
const filter = (value, items) => items.filter((item) => item.sku.includes(value));
```

Reply:
Offline discussion:
1. Work on splitting search into another action so we can search on filters, and filter on searches.
2. We will search through all _visibile_ fields for the user, making the API a single boolean value `search`.